pkg/operator/set_operator_split.go

```diff
@@ -0,0 +1,202 @@
+			return SetOperatorSplit(cCtx, p)
+		},
+		After: telemetry.AfterRunAction(),
+		Flags: getSetOperatorSplitFlags(),
+	}
+
+	return operatorSplitCmd
+}
+
+func SetOperatorSplit(cCtx *cli.Context, p utils.Prompter) error {
+	ctx := cCtx.Context
+	logger := common.GetLogger(cCtx)
+
+	config, err := readAndValidateSetOperatorSplitConfig(cCtx, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read and validate operator split config", err)
+	}
+
+	cCtx.App.Metadata["network"] = config.ChainID.String()
+
+	ethClient, err := ethclient.Dial(config.RPCUrl)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to create new eth client", err)
+	}
+
+	if config.Broadcast {
+
+		eLWriter, err := common.GetELWriter(
+			config.OperatorAddress,
+			config.SignerConfig,
+			ethClient,
+			elcontracts.Config{
+				RewardsCoordinatorAddress: config.RewardsCoordinatorAddress,
+			},
+			p,
+			config.ChainID,
+			logger,
+		)
+
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to get EL writer", err)
+		}
+
+		logger.Infof("Broadcasting set operator transaction...")
+
+		receipt, err := eLWriter.SetOperatorAVSSplit(ctx, config.OperatorAddress, config.AVSAddress, config.Split, true)
+
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to process claim", err)
+		}
+
+		logger.Infof("Set operator transaction submitted successfully")
+		common.PrintTransactionInfo(receipt.TxHash.String(), config.ChainID)
+	} else {
+		noSendTxOpts := common.GetNoSendTxOpts(config.OperatorAddress)
+		_, _, contractBindings, err := elcontracts.BuildClients(elcontracts.Config{
+			RewardsCoordinatorAddress: config.RewardsCoordinatorAddress,
+		}, ethClient, nil, logger, nil)
+		if err != nil {
+			return err
+		}
+
+		code, err := ethClient.CodeAt(ctx, config.OperatorAddress, nil)
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to get code at address", err)
+		}
+		if len(code) > 0 {
+			// Operator is a smart contract
+			noSendTxOpts.GasLimit = 150_000
+		}
+
+		unsignedTx, err := contractBindings.RewardsCoordinator.SetOperatorAVSSplit(noSendTxOpts, config.OperatorAddress, config.AVSAddress, config.Split)
+
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to create unsigned tx", err)
+		}
+		if config.OutputType == string(common.OutputType_Calldata) {
+			calldataHex := gethcommon.Bytes2Hex(unsignedTx.Data())
+
+			if !common.IsEmptyString(config.OutputFile) {
+				err = common.WriteToFile([]byte(calldataHex), config.OutputFile)
+				if err != nil {
+					return err
+				}
+				logger.Infof("Call data written to file: %s", config.OutputFile)
+			} else {
+				fmt.Println(calldataHex)
+			}
+		} else {
+			logger.Infof("This transaction would set the operator split to %d", config.Split)
+		}
+
+		if !config.IsSilent {
+			txFeeDetails := common.GetTxFeeDetails(unsignedTx)
+			fmt.Println()
+			txFeeDetails.Print()
+		}
+
+		fmt.Println("To broadcast the operator set split, use the --broadcast flag")
```

Comment:
this should be inside `!config.IsSilent` code - since with call data they can directly pipe